StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1203CodeFixProvider.cs

```diff
@@ -0,0 +1,95 @@
+    using System.Composition;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CodeActions;
+    using Microsoft.CodeAnalysis.CodeFixes;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+    using StyleCop.Analyzers.Helpers;
+
+    /// <summary>
+    /// Implements code fixes for <see cref="SA1203ConstantsMustAppearBeforeFields"/>.
+    /// </summary>
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(SA1203CodeFixProvider))]
+    [Shared]
+    public class SA1203CodeFixProvider : CodeFixProvider
+    {
+        private static readonly ImmutableArray<string> FixableDiagnostics =
+            ImmutableArray.Create(SA1203ConstantsMustAppearBeforeFields.DiagnosticId);
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return WellKnownFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(CodeAction.Create(OrderingResources.SA1203CodeFix, token => GetTransformedDocumentAsync(context.Document, diagnostic, token), equivalenceKey: nameof(SA1203CodeFixProvider)), diagnostic);
+            }
+
+            return SpecializedTasks.CompletedTask;
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            while (true)
+            {
+                var typeDeclarationNode = syntaxRoot.FindNode(diagnostic.Location.SourceSpan).Ancestors().OfType<TypeDeclarationSyntax>().First();
+                if (typeDeclarationNode == null)
+                {
+                    return document;
+                }
+
+                var allFields = typeDeclarationNode.Members.OfType<FieldDeclarationSyntax>().ToArray();
+                var fieldReplaced = false;
+
+                FieldDeclarationSyntax firstNonConst = null;
+                foreach (var field in allFields)
+                {
+                    if (firstNonConst != null && field.Modifiers.Any(m => m.IsKind(SyntaxKind.ConstKeyword)))
+                    {
+                        syntaxRoot = MoveField(syntaxRoot, field, firstNonConst);
+                        fieldReplaced = true;
+                        break;
+                    }
+
+                    if (firstNonConst == null && !field.Modifiers.Any(m => m.IsKind(SyntaxKind.ConstKeyword)))
```

Comment:
`field.Modifiers.Any(m => m.IsKind(SyntaxKind.ConstKeyword)`
could be computed once